pkg/access_token/access_token_file_test.go

```diff
@@ -19,138 +19,223 @@ func TestAccessTokenFile(t *testing.T) {
-			eq := reflect.DeepEqual(dataActual, dataExpected)
+			Convey("Finishes without raising an error", func() {
+				So(err, ShouldEqual, nil)
+			})
+
+			Convey("When running Read method", func() {
+				dataExpected, err := tokenInFile.Read()
 
-			So(err, ShouldEqual, nil)
-			So(eq, ShouldEqual, true)
+				Convey("Finishes without raising an error", func() {
+					So(err, ShouldEqual, nil)
+				})
+
+				Convey("Returns the data that was written", func() {
+					eq := reflect.DeepEqual(dataActual, dataExpected)
+					So(eq, ShouldEqual, true)
+				})
+			})
 		})
 
-		Convey("Returns error if Data is nil", func() {
+		Convey("Given an access token's data is empty", func () {
 			tokenInFile.Data = nil
-			_, err := tokenInFile.Read()
 
-			So(err.Error(), ShouldEqual, "error reading access token, reason: data is empty")
+			Convey("When running the Read method", func() {
+				_, err := tokenInFile.Read()
+
+				Convey("Raises an error that the data is empty", func() {
+					So(err.Error(), ShouldEqual, "error reading access token, reason: data is empty")
+				})
+			})
 		})
 	})
 
 	Convey("Write", t, func() {
-		Convey("Returns no error if Data input is not nil", func() {
+
+		Convey("Given an access token with data", func() {
 			dataActual := []byte{'t', 'e', 's', 't'}
-			tokenInFile.TokenFilePath = "/tmp/accessTokenFileWrite1"
-			err := tokenInFile.Write(dataActual)
 
-			So(err, ShouldEqual, nil)
-			// Check if file exits
-			_, err = os.Stat("/tmp/accessTokenFileWrite1")
-			So(err, ShouldEqual, nil)
+			Convey("When running the Write method", func() {
+				tokenInFile.TokenFilePath = "/tmp/accessTokenFileWrite1"
+				err := tokenInFile.Write(dataActual)
+
+				Convey("Finishes without raising an error", func() {
+					So(err, ShouldEqual, nil)
+				})
+
+				Convey("Checks that the file exists in the path defined", func (){
+					// Check if file exists
+					_, err = os.Stat("/tmp/accessTokenFileWrite1")
+					So(err, ShouldEqual, nil)
+				})
+
+				Convey("When running Read method", func() {
+					dataExpected, _ := tokenInFile.Read()
+
+					// Confirm data was written
+					Convey("Returns the data the was written to the file", func() {
+						eq := reflect.DeepEqual(dataActual, dataExpected)
+						So(eq, ShouldEqual, true)
+					})
+				})
+			})
 		})
 
-		Convey("Returns error if Data input is nil", func() {
-			err := tokenInFile.Write(nil)
+		Convey("Given an access token without data", func() {
+
+			Convey("When running the Write method", func() {
+				err := tokenInFile.Write(nil)
 
-			So(err.Error(), ShouldEqual, "error writing access token, reason: data is empty")
+				Convey("Raises an error that the access token data is empty", func() {
+					So(err.Error(), ShouldEqual, "error writing access token, reason: data is empty")
+				})
+			})
 		})
 
-		Convey("Returns no error if file already exists", func() {
+		Convey("Given access token with data and a file location defined", func() {
```

Comment:
what's the difference between this sceanrio and the one above? Can't we just add the "When running the Write method a second time" convey in the scenario above?

Reply:
joining those tests now and deleting the latter one since they basically the same, just one with an added complexity